File: scripts/commands/generic/removeFriend.py
from intents.chat import ChatBroadcastIntent
from resources.encodables import ProsePackage
from intents.chat import ChatAvatarRequestIntent
from intents.chat.ChatAvatarRequestIntent.RequestType import FRIEND_REMOVE_TARGET

def execute(galacticManager, player, target, args):
	ghost = player.getPlayerObject()

	if ghost is None or not args:
		return

	name = str(args.split(" ")[0]).lower()

	if not name:
		return

	if ghost.getFriendsList().contains(name) is False:
		ChatBroadcastIntent(player, ProsePackage("@cmnty:friend_not_found", "TT", name)).broadcast()
		return

	ChatAvatarRequestIntent(player, name, FRIEND_REMOVE_TARGET).broadcast()
	return

File: scripts/commands/generic/removeFriend.js
var execute = function(objManager, player, target, args) {
	var ChatAvatarRequestIntent = Java.type("intents.chat.ChatAvatarRequestIntent");
	var RequestType = Java.type("intents.chat.ChatAvatarRequestIntent.RequestType");
	var ProsePackage = Java.type("resources.encodables.ProsePackage");
	var ChatBroadcastIntent = Java.type("intents.chat.ChatBroadcastIntent");
	var ghost = player.getPlayerObject();
	var name;
	
	if(ghost == null || args == null) {
		return;
	}
	
	name = args.split(" ")[0].toLowerCase(java.util.Locale.English);
	
	if(name == null) {
		return;
	}
	
	if(!ghost.getFriendsList().contains(name)) {
		new ChatBroadcastIntent(player, new ProsePackage("@cmnty:friend_not_found", "TT", name)).broadcast();
		return;
	}
	
	new ChatAvatarRequestIntent(player, name, RequestType.FRIEND_REMOVE_TARGET).broadcast();
}